src/rgw/rgw_rest_sts.cc

```diff
@@ -230,6 +235,18 @@ int RGWSTSAssumeRoleWithWebIdentity::get_params()
   sub = s->info.args.get("sub");
   aud = s->info.args.get("aud");
 
+  if (! duration.empty()) {
```

Comment:
AssumeRoleRequestBase constructor already has the logic to get the duration. It uses stoull though. That is also used in both AssumeRole and AssumeRoleWithWebIdentity.